BI0000 when using [Wrap] on property getters and setters

### Steps to Reproduce

Try bind something like this:

```csharp
SemanticColorScheme SemanticColorScheme {
	[Wrap ("Runtime.GetNSObject<SemanticColorScheme> (ColorScheme.Handle, false)")] get;
	[Wrap ("ColorScheme = value")] set;
}
```

Full type:

```csharp
	[BaseType (typeof (NSObject), Name = "MDCCardScheme")]
	interface CardScheme : CardScheming {
		[Export ("colorScheme", ArgumentSemantic.Assign)]
		new IColorScheming ColorScheme { get; set; }

		[Export ("shapeScheme", ArgumentSemantic.Assign)]
		new IShapeScheming ShapeScheme { get; set; }

		SemanticColorScheme SemanticColorScheme {
			[Wrap ("Runtime.GetNSObject<SemanticColorScheme> (ColorScheme.Handle, false)")] get;
			[Wrap ("ColorScheme = value")] set;
		}
	}
```

### Expected Behavior

No error.

### Actual Behavior

💥 

### Environment

https://gist.github.com/mattleibow/953050a4f599d3f7222633939786ad52

### Build Logs

```

Building MaterialComponents (Debug|iPhoneSimulator)
__________________________________________________
Project "/Users/matthew/Projects/XamarinComponents/iOS/MaterialComponents/source/MaterialComponents/MaterialComponents.csproj" (Build target(s)):

/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/bin/btouch-native /v /nostdlib /baselib:/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.iOS/Xamarin.iOS.dll /lib:/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.iOS /r:/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.iOS/mscorlib.dll /unsafe /d:"__UNIFIED__;__MOBILE__;__IOS__;DEBUG" /Users/matthew/Projects/XamarinComponents/iOS/MaterialComponents/source/MaterialComponents/ApiDefinition.cs /s:/Users/matthew/Projects/XamarinComponents/iOS/MaterialComponents/source/MaterialComponents/Structs.cs -r /Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.iOS/mscorlib.dll -r /Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.iOS/System.dll -r /Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.iOS/Xamarin.iOS.dll /tmpdir:/Users/matthew/Projects/XamarinComponents/iOS/MaterialComponents/source/MaterialComponents/obj/Debug/ios/ /sourceonly:/Users/matthew/Projects/XamarinComponents/iOS/MaterialComponents/source/MaterialComponents/obj/Debug/ios/sources.list /target-framework=Xamarin.iOS,v1.0 
ApiDefinition.cs(1540,21): warning CS0618: 'FlatButton' is obsolete: 'This class will be deprecated soon. Consider using TextButtonThemer with a Button instead.'
BTOUCH : error BI0000: Unexpected error - Please file a bug report at http://bugzilla.xamarin.com
Done building project "MaterialComponents.csproj" -- FAILED.

========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========

Build: 1 error, 1 warning
```

### Example Project (If Possible)

<!--
1. Place cursor below this comment block.
2. Drag and drop the compressed project or files needed to reproduce.
-->

<!--
Switch to the "Preview" tab to ensure your issue renders correctly.
-->

--- a/src/generator.cs
+++ b/src/generator.cs
@@ -4526,7 +4526,7 @@ public partial class Generator : IMemberGatherer {
 	
 	bool DoesPropertyNeedDirtyCheck (PropertyInfo pi, ExportAttribute ea) 
 	{
-		switch (ea.ArgumentSemantic) {
+		switch (ea?.ArgumentSemantic) {
 		case ArgumentSemantic.Copy:
 		case ArgumentSemantic.Retain: // same as Strong
 		case ArgumentSemantic.None:

--- a/tests/generator/BGenTests.cs
+++ b/tests/generator/BGenTests.cs
@@ -561,6 +561,9 @@ namespace GeneratorTests
 			Assert.AreEqual (modelName, attrib.ConstructorArguments [0].Value, "Custom ObjC name");
 		}
 
+		[Test]
+		public void GHIssue5444 () => BuildFile (Profile.iOS, "ghissue5444.cs");
+
 		BGenTool BuildFile (Profile profile, params string [] filenames)
 		{
 			return BuildFile (profile, true, false, filenames);

--- a/tests/generator/ghissue5444.cs
+++ b/tests/generator/ghissue5444.cs
@@ -1,0 +1,19 @@
+using Foundation;
+using ObjCRuntime;
+
+namespace GHIssue5444 {
+
+	[BaseType (typeof (NSObject))]
+	interface CardScheme {
+		[Export ("colorScheme", ArgumentSemantic.Assign)]
+		new NSString ColorScheme { get; set; }
+
+		[Export ("shapeScheme", ArgumentSemantic.Assign)]
+		new NSString ShapeScheme { get; set; }
+
+		NSString SemanticColorScheme {
+			[Wrap ("Runtime.GetNSObject<NSString> (ColorScheme.Handle, false)")] get;
+			[Wrap ("ColorScheme = value")] set;
+		}
+	}
+}

